src/Compilers/Core/Portable/Syntax/SyntaxTriviaList.cs

```diff
@@ -351,7 +353,7 @@ public SyntaxTriviaList RemoveAt(int index)
 
             var list = this.ToList();
             list.RemoveAt(index);
-            return new SyntaxTriviaList(default(SyntaxToken), Node.CreateList(list.Select(n => n.UnderlyingNode)), 0, 0);
+            return new SyntaxTriviaList(default(SyntaxToken), Node!.CreateList(list.Select(n => n.RequiredUnderlyingNode)), 0, 0);
```

Comment:
>Node! [](start = 62, length = 5)

nit: Should this be an assertion?

Reply:
The code does validate that `Node` is not null but it's really subtle. Basically for the code to reach this point `Count` must be greater than zero. In that case `Node` is guaranteed to be non-null. 

I'm open to how to make this clearer: either by using `Debug.Assert` or by adding a comment. I agree it's really subtle.